doc/developer/observing_materialize.md

```diff
@@ -0,0 +1,75 @@
+# Observing Materialize
+
+This document covers how to peer into the internals of a Materialize instance to understand what
+it's doing and how it's performing. The intended audience for this document are developers working
+on Materialize and others interested in improving the performance of materialized.
+
+## Sources of Metrics
+
+### Prometheus Metrics
+
+Operational metrics can be scraped by Prometheus from port `6875` under the `/metrics` URL. For
```

Comment:
nit: technically `/metrics` is the path portion of the url.

Reply:
Yes, thank you!